src/profile/profile.controller.ts

```diff
@@ -45,58 +47,46 @@ export class ProfileController {
     return ZaLaResponse.Ok(userProfile, 'Profile created', 201);
   }
 
-  @Get('/:profileId')
-  @IdCheck('profileId')
+  @Get()
   @ApiOperation({ summary: 'Get a profile' })
-  async getOne(
-    @Param('profileId', new ParseUUIDPipe()) profileId: string,
-  ): Promise<Ok<Profile>> {
-    const profile = await this.profileService.getone(profileId);
+  async getOne(@Req() req: Request): Promise<Ok<Profile>> {
+    const profile = await this.profileService.getone(req.profileId);
```

Comment:
I think you should understand what profileId is used for before editing it, e.g here it doesn't make sense to get your own profile.

The aim here is to get any profile using the Id